src/ng/directive/ngTranscludeLocals.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name ngTranscludeLocals
+ * @restrict A
+ *
+ * @description
+ * Directive that allows to define local variables for a given transclusion scope.
+ *
+ * It works with {@link ng.directive:ngTransclude ngTransclude} and it
+ * expects an object which define the local varibles for the transcluded template.
+ *
+ * @element ng-transclude
+ *
+ * @param {expression} ngTranscludeLocals an object with pairs of key-value
+ *                                        in which the key is the variable name inside the template
+ *                                        and the value is the value for those variables
+ *
+ * @example
+ * ### Transclude with locals
+ * This example demonstrates basic transclusion of local context into a transcluded template.
+ * <example name="simpleTranscludeExample" module="transcludeLocalsExample">
+ *   <file name="index.html">
+ *     <script>
```

Comment:
Mainly because I copy pasted the example from other directive.
But, I was trying to make easier to understand, but I think that did not.